tests/Tokenizer/Transformer/TypeIntersectionTransformerTest.php

```diff
@@ -323,4 +324,100 @@ function f( #[Target(\'a\')] #[Target(\'b\')] #[Target(\'c\')] #[Target(\'d\')]
             ],
         ];
     }
+
+    /**
+     * @param array<int, int> $expectedTokens
+     *
+     * @dataProvider provideProcess82Cases
+     *
+     * @requires PHP 8.2
+     */
+    public function testProcess82(string $source, array $expectedTokens): void
+    {
+        $this->doTest($source, $expectedTokens);
+    }
+
+    public static function provideProcess82Cases(): iterable
```

Comment:
Maybe some test case with arrow functions too (argument, return type)? The same in `TypeAlternationTransformerTest`.